pkg/controller/gittrackobject/gittrackobject_controller_test.go

```diff
@@ -511,348 +537,474 @@ var _ = Describe("GitTrackObject Suite", func() {
+						Eventually(requests, timeout).Should(Receive(Equal(expectedClusterRequest)))
+						m.Get(gto, timeout).Should(Succeed())
 					})
 
-					PIt("to the namespace the controller is restricted to", func() {
-						for range events.Items {
-							event := <-testEvents
-							Expect(event.Namespace).To(Equal(farosflags.Namespace))
-						}
+					Context("should update the status", func() {
+						It("to represent the failure", func() {
+							m.Eventually(gto, timeout).Should(
+								testutils.WithGitTrackObjectStatusConditions(
+									ContainElement(
+										SatisfyAll(
+											testutils.WithGitTrackObjectConditionType(Equal(farosv1alpha1.ObjectInSyncType)),
+											testutils.WithGitTrackObjectConditionStatus(Equal(corev1.ConditionFalse)),
+											testutils.WithGitTrackObjectConditionReason(Equal(string(gittrackobjectutils.ErrorUnmarshallingData))),
+										),
+									),
+								),
+							)
+						})
 					})
-				})
-			})
-
-			Context("with invalid data", func() {
-				BeforeEach(func() {
-					// Break the JSON data
-					gto.Spec.Data = gto.Spec.Data[10:]
 
-					// Create and fetch the instance to make sure caches are synced
-					m.Create(gto).Should(Succeed())
-					// Wait twice for the extra reconcile for status updates
-					Eventually(requests, timeout).Should(Receive(Equal(expectedClusterRequest)))
-					Eventually(requests, timeout).Should(Receive(Equal(expectedClusterRequest)))
-					m.Get(gto, timeout).Should(Succeed())
-				})
+					Context("should send an event", func() {
+						var events *corev1.EventList
+						BeforeEach(func() {
+							events = &corev1.EventList{}
+							m.Eventually(events, timeout).ShouldNot(testutils.WithItems(BeEmpty()))
+						})
 
-				Context("should update the status", func() {
-					It("to represent the failure", func() {
-						m.Eventually(gto, timeout).Should(
-							testutils.WithGitTrackObjectStatusConditions(
-								ContainElement(
-									SatisfyAll(
-										testutils.WithGitTrackObjectConditionType(Equal(farosv1alpha1.ObjectInSyncType)),
-										testutils.WithGitTrackObjectConditionStatus(Equal(corev1.ConditionFalse)),
-										testutils.WithGitTrackObjectConditionReason(Equal(string(gittrackobjectutils.ErrorUnmarshallingData))),
-									),
+						It("to represent the failure", func() {
+							m.Eventually(events, timeout).Should(testutils.WithItems(ContainElement(
+								SatisfyAll(
+									testutils.WithReason(Equal("UnmarshalFailed")),
+									testutils.WithInvolvedObjectKind(Equal(gto.GetObjectKind().GroupVersionKind().Kind)),
+									testutils.WithInvolvedObjectName(Equal(gto.GetName())),
+									testutils.WithEventType(Equal(string(corev1.EventTypeWarning))),
 								),
-							),
-						)
+							)))
+						})
 					})
 				})
 
-				Context("should send an event", func() {
-					var events *corev1.EventList
+				// don't run these tests, since the cross-namespace ownership reference is
```

Comment:
I would assume these should either be deleted if they are invalid or rewritten so that they work such that a controller doesn't reconcile a GTO owned by a CGT? Do we have tests for that already?

Reply:
I've added a test for the latter. I kept the test around because while they don't "work", they can be used to figure out what exactly we do when we get invalid k8s data, but I'm gonna remove them anyway